docs/src/components/VersionDropdown.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+
+const CACHE_KEY = 'kargo-docs-versions';
+const CACHE_DURATION = 1000 * 60 * 30; // 30 minutes
+const githubApiUrl = 'https://api.github.com/repos/akuity/kargo/branches?protected=true';
+
+function VersionDropdown() {
+  console.log("Navbar dropdown initialized");
+  const [versions, setVersions] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [currentVersion, setCurrentVersion] = useState("");
+
+  useEffect(() => {
+    fetchVersions();
+  }, []);
+
+  const fetchVersions = async () => {
+    const cachedData = localStorage.getItem(CACHE_KEY);
+    if (cachedData) {
+      const { versions, timestamp } = JSON.parse(cachedData);
+      if (Date.now() - timestamp < CACHE_DURATION) {
+        console.log("Using cached versions:", versions);
+        setVersions(versions);
+        setLoading(false);
+        return;
+      }
+    }
+    try {
+      console.log("Before fetching versions")
+      const response = await fetch(githubApiUrl);
+      const branches = await response.json();
+      console.log("fetched branches are: ", branches);
+
+      const releaseBranches = branches
+        .map(branch => branch.name)
+        .filter(name => /^release-([1-9]\d*\.\d+)$/.test(name))
+        .map(name => {
+          const [major, minor] = name.replace('release-', '').split('.');
+          return {
+            version: `v${major}.${minor}`,
+            url: `https://${name.replace('.', '-')}.docs.kargo.io`
+          };
+        });
+        console.log("These are release branches before unshifting: ", releaseBranches)
+      // Add current version
+      releaseBranches.unshift({
+        version: 'Latest Version',
+        url: `https://docs.kargo.io`
+      });
```

Comment:
```suggestion
      console.log("These are release branches before sorting and updating 0 element: ", releaseBranches)
      releaseBranches.sort((a, b) => {
        if (a.version > b.version) {
          return -1;
        }
        if (a.version < b.version) {
          return 1;
        }
        return 0;
      });
      // Overwrite the first element with the latest version
      releaseBranches[0] = {
        version: 'Latest Version',
        url: `https://docs.kargo.io`
      };
```

Reply:
I think you're also going to have to look at `window.location` to get the domain and determine which version is currently selected. As it is right now, you will see `Latest Version` in your navbar no matter which one you're actually looking at.